query-engine/connectors/sql-query-connector/src/query_builder/select.rs

```diff
@@ -152,8 +153,9 @@ impl SelectBuilder {
 
     fn build_m2m_join<'a>(&mut self, rs: &RelationSelection, parent_alias: Alias, ctx: &Context<'_>) -> JoinData<'a> {
         let rf = rs.field.clone();
-        let m2m_alias = m2m_join_alias_name(&rf);
         let m2m_table_alias = self.next_alias();
```

Comment:
We know use `Alias` instead of the generated string. This is to enable passing the alias to the `FilterVisitor` which expects an `Alias` and not a random string.